### PR TITLE
os/bluestore: misc fixes

### PR DESCRIPTION
**Pull request by xiexingguo**
We shall round nbytes up to be page-size(4096) aligned instead of
truncating it down, as syncing more is usually okay
while syncing less indicates potential data loss.

Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

### CODE REVIEW

**Comment by liewegas** [comment]
I think this is a poorly named function.. the comment says

   // This asks the OS to initiate flushing the cached data to disk,
   // without waiting for completion.

so RangeStartFlush would have been better.  In any case, it's not a data integrity call, so I think it's okay as it.

What I want to avoid is flushing the same block twice...

**Comment by xiexingguo** [comment]
@liewegas Dropped that change and appended more changes.

File: src/os/bluestore/BlueFS.cc [hunk 1]
```diff
@@ -1379,8 +1379,9 @@ int BlueFS::_flush_range(FileWriter *h, uint64_t offset, uint64_t length)
 		      offset + length - allocated,
 		      &h->file->fnode.extents);
     if (r < 0) {
-      derr << __func__ << " allocated: " << allocated << \
-               " offset: " << offset << " length: " << length << dendl;
+      derr << __func__ << " allocated: 0x" << std::hex << allocated
+           << " offset: 0x" << offset << " length: 0x" << length << std::dec
```
Comment (liewegas) [inline src/os/bluestore/BlueFS.cc]: please include the std::dec at the end

**Comment by liewegas** [comment]
otherwise lgtm, thanks!

File: src/os/bluestore/BlueFS.cc (hunk 1 above)
Comment (xiexingguo) [inline src/os/bluestore/BlueFS.cc]: @liewegas Sir, I did... The `std::dec` lies at the end of this line.

Comment (liewegas) [inline src/os/bluestore/BlueFS.cc]: Oops, yes, sorry about that!

**Pull request merged**